piptools/scripts/compile.py

```diff
@@ -291,6 +305,27 @@ def _determine_linesep(
     help="Specify a package to consider unsafe; may be used more than once. "
     f"Replaces default unsafe packages: {', '.join(sorted(UNSAFE_PACKAGES))}",
 )
+@click.option(
+    "--build-distribution",
+    "build_distributions",
+    multiple=True,
+    type=click.Choice(("editable", "sdist", "wheel")),
+    help="Name of a distribution to install build dependencies for; may be used more than once. "
+    "Static dependencies declared in pyproject.toml will be included as well.",
+)
+@click.option(
+    "--all-build-distributions",
+    is_flag=True,
+    default=False,
+    help="Pin build dependencies requested by any of the build backend hooks. "
+    "Static dependencies declared in pyproject.toml will be included as well.",
+)
+@click.option(
+    "--only-build-distributions",
+    is_flag=True,
+    default=False,
+    help="Exclude dependencies other than build dependencies.",
```

Comment:
"exclude" does not really relay what'd happen if the same dist is both a build and a runtime dep. Maybe rephrase?

Reply:
Is the new phrasing better?

Would it be even better to change this option to something like `--no-default-sources` having the effect that only explicitly included sources such as extras and build deps are considered?

Reply:
Marking this as resolved since it is now hard to track in the code and I have made an attempt at addressing this.